airbyte-cdk/python/docs/tutorials/cdk-tutorial-any-percent/cdk-speedrun.md

```diff
@@ -0,0 +1,220 @@
+    "properties": {
+      "start_date": {
+        "type": "string",
+        "description": "Start getting data from that date.",
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
+        "examples": ["%Y-%m-%d"]
+      },
+      "base": {
+        "type": "string",
+        "examples": ["USD", "EUR"],
+        "description": "ISO reference currency. See <a href=\"https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/index.en.html\">here</a>."
+      }
+    }
+  }
+}
+```
+
+Ok, let's write a function that checks the inputs we just defined. Nuke the `source.py` file. Now add this code to it. For a crucial time skip, we're going to define all the imports we need in the future here.
+
+```python
+from datetime import datetime, timedelta
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import NoAuth
+
+class SourcePythonHttpTutorial(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        accepted_currencies = {
+            "USD",
+            "JPY",
+            "BGN",
+            "CZK",
+            "DKK",
+        }  # there are more currencies but let's assume these are the only allowed ones
+        input_currency = config["base"]
+        if input_currency not in accepted_currencies:
+            return False, f"Input currency {input_currency} is invalid. Please input one of the following currencies: {accepted_currencies}"
+        else:
+            return True, None
+        
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        # Parse the date from a string into a datetime object
+        start_date = datetime.strptime(config["start_date"], "%Y-%m-%d")
+        return [ExchangeRates(authenticator=auth, base=config["base"], start_date=start_date)]
+```
+
+Test it.
+
+```bash
+cd ..
+mkdir sample_files
+echo '{"start_date": "2021-04-01", "base": "USD"}'  > sample_files/config.json
+echo '{"start_date": "2021-04-01", "base": "BTC"}'  > sample_files/invalid_config.json
+python main.py check --config sample_files/config.json
+python main.py check --config sample_files/invalid_config.json
+```
+
+Expected output: 
+```text
+> python main.py check --config sample_files/config.json
+{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "SUCCEEDED"}}
+
+> python main.py check --config sample_files/invalid_config.json
+{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "Input currency BTC is invalid. Please input one of the following currencies: {'DKK', 'USD', 'CZK', 'BGN', 'JPY'}"}}
+```
+
+## Define your Stream
+
+In your `source.py` file, add this `PythonHttpExampleStream` class. This stream represents an endpoint you want to hit.
```

Comment:
This is not the name of the class
```suggestion
In your `source.py` file, add this `ExchangeRates` class. This stream represents an endpoint you want to hit.
```

Reply:
There's a couple of references to this throughout so you might wanna double check. I just did a speedrun review